numpy/core/src/multiarray/item_selection.c

```diff
@@ -1292,7 +1292,14 @@ partition_prep_kth_array(PyArrayObject * ktharray,
     npy_intp * kth;
     npy_intp nkth, i;
 
-    if (!PyArray_CanCastSafely(PyArray_TYPE(ktharray), NPY_INTP)) {
+    if (PyArray_ISBOOL(ktharray)) {
+        /* 2021-09-29, NumPy 1.22 */
+        if (DEPRECATE(
+                "Passing booleans as partition index is deprecated") < 0) {
```

Comment:
Can you add the NumPy version in the message?  Just for completeness sake (could just be a parantheses at the end, I guess, like `(warning added in NumPy 1.22)`.  I know that most older warnings don't have it, but it is nice, I think.

Reply:
Sure thing; done as of https://github.com/numpy/numpy/pull/20000/commits/fd00a92a0d4793076533bf4cef23a1876fadf0fb.